front/src/modules/simulationResult/components/SpaceTimeChartWrapper/SpaceTimeChartWrapper.tsx

```diff
@@ -211,10 +211,70 @@ const SpaceTimeChartWrapper = ({
     allTrainsProjected
   );
 
-  const splitPoints = useMemo<SplitPoint[]>(() => {
-    const pathsIndex = keyBy(paths, ({ id }) => id);
+  const pathsById = useMemo(() => keyBy(paths, ({ id }) => id), [paths]);
 
-    return (
+  const selectedTimetableOccurrencesNb = useMemo(() => {
+    const pacedId = isOccurrenceId(selectedProjectionId)
+      ? extractPacedTrainIdFromOccurrenceId(selectedProjectionId)
+      : isPacedTrainId(selectedProjectionId)
+        ? selectedProjectionId
+        : undefined;
+
+    if (!pacedId) return 0;
+
+    return projectedTrains.filter(
+      (train) =>
+        isOccurrenceId(train.id) &&
+        extractPacedTrainIdFromOccurrenceId(train.id) === pacedId &&
+        (train.spaceTimeCurves?.length ?? 0) > 0
+    ).length;
+  }, [projectedTrains, selectedProjectionId]);
+
+  const hoveredPathId = useMemo(() => {
+    const element = hoveredItem?.element;
+    return element && 'pathId' in element ? element.pathId : undefined;
+  }, [hoveredItem]);
+
+  const baseZonesByWaypointId = useMemo(() => {
+    const map = new Map<string, OccupancyZone[]>();
+    for (const waypoint of trackOccupancyDiagramsData || []) {
+      const zones = (waypoint.zones || []).map((zone) => {
+        const path = pathsById[zone.trainId];
+        if (!path) return zone;
+        return path?.label ? { ...zone, trainName: path.label } : zone;
+      });
+      map.set(waypoint.waypointId, zones);
+    }
+    return map;
+  }, [trackOccupancyDiagramsData, pathsById]);
```

Comment:
You don't need pathsById anymore in your depths
```suggestion
  }, [trackOccupancyDiagramsData]);
```